src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,151 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
+    public MyLinkedList() {
+
+    }
+
     @Override
     public void add(T value) {
+        if (head == null) {
+            Node<T> node = new Node<>(null, value, null);
+            head = node;
+            tail = node;
+            size++;
+            return;
+        }
+
+        Node<T> prev = tail;
+        Node<T> node = new Node<>(prev, value, null);
+        tail = node;
+        prev.next = node;
+        size++;
```

Comment:
let's be more concise 
```suggestion
        Node<T> newNode = new Node<>(tail, value, null);
        if (size == 0) {
            head = newNode;
        } else {
            tail.next = newNode;
        }
        tail = newNode;
        size++;
```